symplyphysics/laws/thermodynamics/efficiency_factor.py

```diff
@@ -0,0 +1,33 @@
+gfrom sympy import (Eq, solve)
+from symplyphysics import (units, Quantity, Symbol, print_expression,
+                           validate_input, validate_output, dimensionless)
+
+# Description
+## The efficiency of a heat engine is the ratio of the useful energy used to the total amount of energy received by the system: eta = (Q_h - Q_r) / Q_h.
+## Where:
+## Q_h - the amount of heat transferred by the heater to the heat engine
+## Q_r - the amount of heat transferred by the heat engine to the refrigerator
+## eta - efficiency of the heat engine
+
+heat_from_heater = Symbol("heat_from_heater", units.energy)
+heat_to_refrigerator = Symbol("heat_to_refrigerator", units.energy)
+efficiency_factor = Symbol("efficiency_factor", dimensionless)
+
+law = Eq(efficiency_factor, (heat_from_heater - heat_to_refrigerator) / heat_from_heater)
+
+
+def print_law() -> str:
+    return print_expression(law)
+
+
+@validate_input(heat_from_heater_=heat_from_heater,
+                heat_to_refrigerator_=heat_to_refrigerator)
+@validate_output(efficiency_factor)
+def calculate_efficiency_factor(heat_from_heater_: Quantity,
+                                heat_to_refrigerator_: Quantity) -> Quantity:
```

Comment:
It is preferable to return `float` instead of dimensionless quantity. See how it is done in another law: https://github.com/blackyblack/symplyphysics/blob/17d00b153fc87168dba61bdf9eb94ca53dd2d14f/symplyphysics/definitions/refractive_index_is_wave_speeds_ratio.py#L49